websites/views.py

```diff
@@ -575,7 +575,9 @@ def get_queryset(self):
         if types:
             queryset = queryset.filter(type__in=types)
         if search:
-            queryset = queryset.filter(title__icontains=search)
+            queryset = queryset.filter(
+                Q(title__icontains=search) | Q(file__iregex=f"{search}([^/]+|$)")
```

Comment:
This still needs to be tweaked a bit to:
```
Q(file__iregex=f"{search}([^/]+$|$)
```

For example I have a site with `short_id=new-site-1` and a resource with `file=courses/new-site-1/biology.jpeg`

Searching for `new` with current code returns the above resource (and all other resources for the site, they all have the same path).  With the suggested tweak, it won't be returned, but a resource with `file=courses/new-site-1/biology-new-photo.jpeg` will be.

Reply:
Updated in 929092e405c3630d744d9e3e0b96e1e9129fb513.